File: apps/desktop/src/lib/group.ts
// Mirrors crates/domain/src/group.rs Group's serde output.
export interface Group {
  id: string;
  name: string;
  destination_path: string | null;
  icon: string | null;
  is_pinned: boolean;
  sort_order: number;
  created_at: string;
  updated_at: string;
}

File: crates/domain/src/group.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

/// A user-defined context ("project") a file can be filed into.
/// MVP keeps this single-parent (one file → one primary group).
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Group {
    pub id: Uuid,
    pub name: String,
    pub destination_path: Option<String>,
    pub icon: Option<String>,
    pub is_pinned: bool,
    pub sort_order: i64,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}
